Added build/usage instructions to readme (#42)
wasmfx/fiber-c: README.md

1. Stackless fibers using the Asyncify transform of binaryen.
2. Native stackful fibers powered by the WasmFX instruction set.

## Installation
Clone the repository:
```
git clone https://github.com/wasmfx/fiber-c.git
```

## Building
To compile the benchmarks programs in `examples` from C to Wasm binaries, you will need:

1. [WASI-SDK 30.0](https://github.com/webassembly/wasi-sdk)
2. [Wasm reference interpreter from the stack-switching proposal](https://github.com/WebAssembly/stack-switching/tree/main/interpreter)
3. [Up-to-date Binaryen](https://github.com/WebAssembly/binaryen)
4. A Wasm execution engine with support for the stack switching instruction set, e.g. [Wasmtime](https://github.com/bytecodealliance/wasmtime), [V8 canary](https://chromium.googlesource.com/v8/v8/+/canary), or [Wizard](https://github.com/titzer/wizard-engine).

Please install and build these packages according to their docs. Please also make sure they are installed in the same root directory, 

> then update the `ROOT` field of `make.config`.

Reviewer: Here's my suggested addition, which I think can go right here:

> Instead of installing these dependencies manually, you can use the docker container build at <http://github.com/wasmfx/benchtainer> and run compilation/benchmarking runs within that container. See instructions therein.

1. Stackless fibers using the Asyncify transform of binaryen.
2. Native stackful fibers powered by the WasmFX instruction set.

## Installation
Clone the repository:
```
git clone https://github.com/wasmfx/fiber-c.git
```

## Building
To compile the benchmarks programs in `examples` from C to Wasm binaries, you will need:

1. [WASI-SDK 30.0](https://github.com/webassembly/wasi-sdk)
2. [Wasm reference interpreter from the stack-switching proposal](https://github.com/WebAssembly/stack-switching/tree/main/interpreter)
3. [Up-to-date Binaryen](https://github.com/WebAssembly/binaryen)

> 4. [Wasmfxtime](https://github.com/wasmfx/wasmfxtime), a fork of Wasmtime extended with support for the WasmFX instruction set.

Reviewer 2: ```suggestion
4. A Wasm execution engine with support for the stack switching instruction set, e.g. [Wasmtime](https://github.com/bytecodealliance/wasmtime), [V8 canary](https://chromium.googlesource.com/v8/v8/+/canary), or [Wizard](https://github.com/titzer/wizard-engine).
```

**Note**: You will also need [upstream wasmtime](https://github.com/bytecodealliance/wasmtime) 

> to build the benchmarks targeting the `switch` instruction.

Reviewer 2: ```suggestion
```

Please install and build these packages according to their docs. Please also make sure they are installed in the same root directory, 
then update the `ROOT` field of `make.config`.

After this, you should be able to run
```
make all
```
All binaries and compilation artefacts are stored in the `out` directory.

To discard all existing artefacts, run
```
make clean
```

## Running the benchmarks
The benchmarking script in `bench.py` can build, run, and time each benchmark. To get it working, you will need to:

1. Install and update engine dependencies 
    

>     1. Wasmfxtime

Reviewer 2: ```suggestion
    1. Wasmtime
```

>     1. Wasmtime

Reviewer 2: For consistency:
```suggestion
    1. [Wasmtime](https://github.com/bytecodealliance/wasmtime)
```

Author: Wasmtime is linked to on line 26 already so it's probably okay

Reviewer 2: It is about aesthetics. It is bad style to have one non-hyperlink entry in a list of three where the other two are hyperlinks.

Reviewer: To be fair, in a context like Wikipedia, I think they would link the first use of a concept, and not every use, even if that resulted in a list like this one with some links some plain. So, I can see it either way. 🤷

Reviewer: In this case, we're giving _instructions_ to a person to go install these things, so it would be appropriate to say, "Now click this link and install this thing, … now this one … "

But I'm hoping that folks will use a true script (i.e. the Dockerfile) in any case :D
